src/v/utils/token_bucket.h

```diff
@@ -79,9 +80,23 @@ class token_bucket {
         }
 
         if (_rate > new_rate) {
-            _sem.consume(_rate - new_rate);
+            if (_sem.current() <= _rate) {
+                _sem.consume(_rate - new_rate);
+            } else {
+                /*
+                 * if current > rate it means that we have accumulated burst
+                 * we should change current to new_rate to start accumulation
+                 * from begining
+                 */
+                _sem.consume(_sem.current() - new_rate);
+            }
```

Comment:
i'm a little unsure about why we need to change the algorithm to support burstiness in that i'd expect the number of tokens in the bucket to determine how large a burst will be tolerated.

can you explain the changes to the algorithm in the commit message? for example, what is a burst_rate_multiplier?

Reply:
I think Roman included this in anticipation that it would be leveraged by the KIP-599 work

Reply:
Also we discussed it in RFC presentation.
Decided that we need ability to accept burst requests in controller log limiting also